packages/nextjs/components/scaffold-eth/Contract/WriteOnlyFunctionForm.tsx

```diff
@@ -32,7 +32,6 @@ export const WriteOnlyFunctionForm = ({ abiFunction, onChange, contractAddress }
     isLoading,
     writeAsync,
   } = useContractWrite({
-    chainId: getTargetNetwork().id,
```

Comment:
will it work on other networks?

Reply:
Yup, if `chainId` is not passed viem / wagmi infers chain from walletClient sending the transaction checkout this => https://github.com/wagmi-dev/viem/blob/0e928cfe04407a2c974aa8a925dacc7687e78c32/src/actions/wallet/sendTransaction.ts#L134C26-L134C26

Aslo in Wagmi docs its optional https://wagmi.sh/react/hooks/useContractWrite#chainid-optional

Reply:
Thanks! Approved